### PR TITLE
Fix Modify interaction with hitDetection

### PR DESCRIPTION
**Pull request by M393**
Fixes Modify configured with hit detection.
See #12235

Hit detection should ignore any RenderFeatures and features not in the configured source / collection.

### CODE REVIEW

**Comment by M393** [comment]
This most likely doesn't fix what @mike-000 reported in https://github.com/openlayers/openlayers/issues/12235#issuecomment-825086303

File: src/ol/interaction/Modify.js [hunk 1]
```diff
@@ -1136,12 +1138,16 @@ class Modify extends PointerInteraction {
         pixel,
         (feature, layer, geometry) => {
           geometry = geometry || feature.getGeometry();
-          if (geometry.getType() === GeometryType.POINT) {
+          if (
+            geometry.getType() === GeometryType.POINT &&
+            geometry.getCoordinates && // Skip RenderFeature
+            this.features_.getArray().includes(feature)
```
[the remark this answers is not in the document]
Comment (M393) [reply src/ol/interaction/Modify.js]: Thanks @mike-000, changed it.

**Review by ahocevar — COMMENTED** [review]
I don't think it is necessary to skip render features. `getFlatCoordinates` could be used instead of `getCoordinates` to support both features and render features.

**Comment by ahocevar** [comment]
> This most likely doesn't fix what @mike-000 reported in [#12235 (comment)](https://github.com/openlayers/openlayers/issues/12235#issuecomment-825086303)

That issue is fixed by #12408.

**Review by ahocevar — CHANGES_REQUESTED** [review]
Thanks @M393! Please modify this pull request according to my suggestions - we can easily handle render features too by using `getFlatCoordinates()` instead of `getCoordinates()`.

**Review by ahocevar — APPROVED** [review]
Thanks, @M393

**Pull request merged**